microsoft/testsuites/network/sriov.py

```diff
@@ -542,6 +544,158 @@ def sriov_ethtool_offload_setting(self, environment: Environment) -> None:
         )
         assert_that(iperf_log).does_not_contain("error")
 
+    @TestCaseMetadata(
+        description="""
+        This case is to verify interrupts count increased after network traiffic
+         went through the VFs.
+
+        Steps,
+        1. Start iperf3 on server node.
+        2. Get initial interrupts sum per irq and cpu number on client node.
+        3. Start iperf3 for 5 minutes with 128 threads on client node.
+        4. Get final interrupts sum per irq number on client node.
+        5. Compare interrupts changes, expected to see interrupts increased.
+        6. Get final interrupts sum per cpu on client node.
+        7. Compare interrupts changes, expected to see interrupts increased.
+        """,
+        priority=2,
+        timeout=1200,
+        requirement=node_requirement(
+            node=schema.NodeSpace(
+                node_count=2,
+                core_count=search_space.IntRange(min=8, max=16),
+                network_interface=schema.NetworkInterfaceOptionSettings(
+                    nic_count=search_space.IntRange(min=1, max=8),
+                    data_path=schema.NetworkDataPath.Sriov,
+                ),
+            )
+        ),
+    )
+    def sriov_interrupts_change(self, environment: Environment) -> None:
+        server_node = cast(RemoteNode, environment.nodes[0])
+        client_node = cast(RemoteNode, environment.nodes[1])
+        vm_nics = self._initialize_nic_info(environment)
+
+        server_iperf3 = server_node.tools[Iperf3]
+        client_iperf3 = client_node.tools[Iperf3]
+        # 1. Start iperf3 on server node.
+        server_iperf3.run_as_server()
+
+        for _, client_nic_info in vm_nics[client_node.name].items():
+            # 2. Get initial interrupts sum per irq and cpu number on client node.
+            initial_pci_interrupts_by_irqs = (
+                Interrupt.get_sum_of_interrupt_data_by_irqs(
+                    client_node, client_nic_info.pci_slot
+                )
+            )
+            initial_pci_interrupts_by_cpus = (
+                Interrupt.get_sum_of_interrupt_data_by_cpus(
+                    client_node, client_nic_info.pci_slot
+                )
+            )
+            matched_server_nic_info: NicInfo
+            for _, server_nic_info in vm_nics[server_node.name].items():
+                if (
+                    server_nic_info.ip_addr.rsplit(".", maxsplit=1)[0]
+                    == client_nic_info.ip_addr.rsplit(".", maxsplit=1)[0]
+                ):
+                    matched_server_nic_info = server_nic_info
+                    break
+            assert matched_server_nic_info
+            # 3. Start iperf3 for 10 minutes with 128 threads on client node.
+            client_iperf3.run_as_client(
```

Comment:
How long this test case takes? Can it be in 30 seconds?

Reply:
update it into 60s

Reply:
How about another one?

Reply:
change into 30 secs

Reply:
Another test case is still 1800 seconds.

Reply:
that is stress testing